app/models/feed_token.rb

```diff
@@ -10,7 +10,15 @@ def set_defaults
     self.token ||= SecureRandom.urlsafe_base64(20)
   end
 
-  def feed_published_url_with_token
-    feed.published_url + '?auth=' + token
+  def self.feed_published_url_with_token(some_feed)
+    raise "missing token for private feed" unless some_feed.tokens.any?
+
+    token = some_feed.tokens.first.token
+
+    # use the feed's published_url, but replace the path with the token using substitution
+    some_feed.
+      published_url.
+      sub("{?auth}", "?auth=#{token}").
+      sub("f.prxu.org", "p.prxu.org")
```

Comment:
I don't think this is necessary?

The `feed.published_url` already checks if it's `private?` - and varies the host to `ENV['FEEDER_CDN_PRIVATE_HOST']` if needed.